tmva/tmva/inc/TMVA/DNN/CNN/ConvLayer.h

```diff
@@ -46,21 +46,35 @@ class TConvLayer : public VGeneralLayer<Architecture_t> {
    using Scalar_t = typename Architecture_t::Scalar_t;
 
 private:
+   bool inline isInteger(Scalar_t x) const { return x == floor(x); }
+
+   /* Calculate the output dimension of the convolutional layer */
+   size_t calculateDimension(int imgDim, int fltDim, int padding, int stride);
+
+   /* Calculate the number of pixels in a single receptive field */
+   size_t inline calculateNLocalViewPixels(int depth, int height, int width) { return depth * height * width; }
+
+   /* Calculate the number of receptive fields in an image given the filter and image sizes */
+   size_t calculateNLocalViews(int inputHeight, int filterHeight, int paddingHeight, int strideRows, int inputWidth,
+                               int filterWidth, int paddingWidth, int strideCols);
+
+protected:
    size_t fFilterDepth;  ///< The depth of the filter.
    size_t fFilterHeight; ///< The height of the filter.
    size_t fFilterWidth;  ///< The width of the filter.
 
    size_t fStrideRows; ///< The number of row pixels to slid the filter each step.
    size_t fStrideCols; ///< The number of column pixels to slid the filter each step.
 
-   size_t fPaddingHeight; ///< The number of zero layers added top and bottom of the input.
-   size_t fPaddingWidth;  ///< The number of zero layers left and right of the input.
-
    size_t fNLocalViewPixels; ///< The number of pixels in one local image view.
    size_t fNLocalViews;      ///< The number of local views in one image.
 
    Scalar_t fDropoutProbability; ///< Probability that an input is active.
 
+private:
+   size_t fPaddingHeight; ///< The number of zero layers added top and bottom of the input.
```

Comment:
why now fFilterDepdth, height are now protected and the paddingheight is private ?

Reply:
Everything that is needed on `MaxPoolingLayer` (which is now a subclass of `ConvLayer`) is now protected to reduce duplication. Since pooling uses 0 padding by definition, this field remains private to `ConvLayer`